### PR TITLE
[syncd] Create base class SaiSwitchInterface

### PR DESCRIPTION
**Pull request by kcudnik**
This class will be needed for future asic compare.

Signed-off-by: kcudnik <kcudnik@gmail.com>

### CODE REVIEW

File: syncd/SaiSwitchInterface.h [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                    _In_ sai_object_id_t switchRid);
+
+            virtual ~SaiSwitchInterface() = default;
+
+        public:
+
+            sai_object_id_t getVid() const;
+            sai_object_id_t getRid() const;
+
+        public:
```
Comment (qiluo-msft) [inline syncd/SaiSwitchInterface.h:34]: >public: [](start = 8, length = 7)

Wondering why you need consequent `public:`

File: syncd/SaiSwitchInterface.h [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+            virtual std::unordered_map<sai_object_id_t, sai_object_id_t> getVidToRidMap() const = 0;
+
+            virtual std::unordered_map<sai_object_id_t, sai_object_id_t> getRidToVidMap() const = 0;
+
+            virtual bool isDiscoveredRid(
+                    _In_ sai_object_id_t rid) const  = 0;
```
Comment (qiluo-msft) [inline syncd/SaiSwitchInterface.h:41]: >   [](start = 51, length = 2)

Use one blank?

**Review by qiluo-msft — COMMENTED** [review]
As comments

File: syncd/SaiSwitchInterface.h (hunk 2 above)
Comment (kcudnik) [reply syncd/SaiSwitchInterface.h]: done

File: syncd/SaiSwitchInterface.h (hunk 1 above)
Comment (kcudnik) [reply syncd/SaiSwitchInterface.h:34]: i use this as separator for similar blocks, here block of non virtual public functions , and next block of public virtual functions

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**